metrics/prometheus/reporter.go

```diff
@@ -0,0 +1,85 @@
+package prometheus
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/micro/go-micro/v3/metrics"
+	"github.com/micro/micro/v3/service"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// TODO: These sensible defaults should really become config options
+var (
+	// The Prometheus metrics will be made available on this port:
+	listenAddress = ":9000"
+	// This is the endpoint where the Prometheus metrics will be made available ("/metrics" is the default with Prometheus):
+	metricsEndpoint = "/metrics"
+	// timingObjectives is the default spread of stats we maintain for timings / histograms:
+	timingObjectives = map[float64]float64{0.0: 0, 0.5: 0.05, 0.75: 0.04, 0.90: 0.03, 0.95: 0.02, 0.98: 0.001, 1: 0}
+)
+
+// Reporter is an implementation of metrics.Reporter:
+type Reporter struct {
+	defaultLabels      prometheus.Labels
+	prometheusRegistry *prometheus.Registry
+	counters           metricFamily
+	gauges             metricFamily
+	timings            metricFamily
+}
+
+// New returns a configured prometheus reporter:
+func New(service service.Service) (*Reporter, error) {
```

Comment:
Micro builds on go-micro so theres no back reference here. See other packages where we define an Options struct and then Option func,